### PR TITLE
chore: removing branch protection trigger from git connect flow

### PR DESCRIPTION
**Pull request by brayn003**
## Description
1. Remove branch protection trigger from git connect saga
2. Updating test-cases and removed logic for removing branch protection after git connect

Fixes https://github.com/appsmithorg/appsmith/issues/34059

## Automation

/ok-to-test tags="@tag.Git"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9498573284>
> Commit: 6ef7f8bc0bfbe15880857fff1f6262c9e9b5ef4a
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9498573284&attempt=1" target="_blank">Click here!</a>

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **Bug Fixes**
  - Corrected Git branch protection interactions in test scenarios.

- **New Features**
  - Enhanced Git connection success messages and actions.

- **Improvements**
  - Reordered import statements for better code organization.
  - Adjusted visibility of methods and refactored parameters in Git synchronization.

- **Dependencies**
  - Updated `design-system` dependency to version `2.1.42`.

### CODE REVIEW

File: app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx [hunk 1]
```diff
@@ -35,43 +50,86 @@
...
+        </Text>
+      </div>
+      <LinkText isBold renderAs="p">
+        <Link href={DOCS_BRANCH_PROTECTION_URL}>
+          {createMessage(GIT_CONNECT_SUCCESS_PROTECTION_DOC_CTA)}
+        </Link>
+      </LinkText>
+    </>
+  );
+}
```
[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx:130]: Nope, not part of design

File: app/client/cypress/support/Pages/GitSync.ts [hunk 2]
```diff
@@ -118,7 +118,6 @@ export class GitSync {
     repoName = "Repo",
```
Comment (ayushpahwa) [inline app/client/cypress/support/Pages/GitSync.ts:118]: Would be better to assign all items to public/private explicitly for clarity. In most cases, not all of these variables are to be public.

File: app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx [hunk 3]
```diff
@@ -4,25 +4,40 @@ import {
...
+  Tooltip,
+} from "design-system";
 import React, { useEffect } from "react";
 import { useDispatch, useSelector } from "react-redux";
 import styled from "styled-components";
 import { getCurrentAppGitMetaData } from "@appsmith/selectors/applicationSelectors";
 import AnalyticsUtil from "@appsmith/utils/AnalyticsUtil";
 import { GitSettingsTab } from "reducers/uiReducers/gitSyncReducer";
+import { DOCS_BRANCH_PROTECTION_URL } from "constants/ThirdPartyConstants";
+import { importRemixIcon } from "design-system-old";
```
Comment (ayushpahwa) [inline app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx:33]: Why are we using design-system-old here? We should be adding the icon in design system repo and importing here.

File: app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx [hunk 4]
```diff
@@ -35,43 +50,86 @@ const TitleText = styled(Text)`
...
+const InlineIcon = styled(Icon)`
   display: inline-flex;
-  align-items: center;
-  margin-right: 4px;
 `;
 
-const BranchTag = styled(Tag)`
-  display: inline-flex;
+const DetailContainer = styled.div`
+  width: 172px;
```
Comment (ayushpahwa) [inline app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx:58]: If this is the only css you need, might be a better option to use the `Flex` component from design system or using tailwind. Styled classes make it hard to read the code unless they don't offer much abstraction.

File: app/client/cypress/support/Pages/GitSync.ts (hunk 2 above)
Comment (brayn003) [reply app/client/cypress/support/Pages/GitSync.ts:118]: All the items are already explicitly marked as private/public.
Also, we are planning to pick up refactoring tests in future sprints. It is not in the scope for this PR

**Review by ayushpahwa — APPROVED** [review]

**Pull request merged**